File: src/test/java/cn/enaium/cf4m/test/event/EventCancelTest.java
package cn.enaium.cf4m.test.event;

import cn.enaium.cf4m.CF4M;
import cn.enaium.cf4m.annotation.Event;
import cn.enaium.cf4m.test.event.event.EventCancelEventTest;

/**
 * @author Enaium
 */
public class EventCancelTest {
    public EventCancelTest() {
        CF4M.EVENT.register(this);
        EventCancelEventTest eventCancelEventTest = new EventCancelEventTest();
        if (!eventCancelEventTest.cancel) {
            CF4M.EVENT.post(eventCancelEventTest);
        }
        if (!eventCancelEventTest.cancel) {
            CF4M.EVENT.post(eventCancelEventTest);
        }
    }

    @Event
    public void onUpdate(EventCancelEventTest event) {
        event.cancel = true;
        System.out.println("Event");
    }
}

File: src/test/java/cn/enaium/cf4m/test/event/EventCancelTest.kt
package cn.enaium.cf4m.test.event

import cn.enaium.cf4m.test.event.event.EventCancelEventTest
import cn.enaium.cf4m.CF4M.INSTANCE
import cn.enaium.cf4m.annotation.Event

/**
 * Project: cf4m
 *
 * @author Enaium
 */
class EventCancelTest {
    @Event
    fun onUpdate(event: EventCancelEventTest) {
        event.cancel = true
        println("Event")
    }

    init {
        INSTANCE.event.register(this)
        val eventCancelEventTest = EventCancelEventTest()
        if (!eventCancelEventTest.cancel) {
            INSTANCE.event.call(eventCancelEventTest);
        }
        if (!eventCancelEventTest.cancel) {
            INSTANCE.event.call(eventCancelEventTest);
        }
    }
}